content/learning-paths/cross-platform/integer-vs-floats/implicit-conversions.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Implicit conversions, in detail
+weight: 4
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+## A closer look at implicit conversions
+
+Let's take a small mathematical problem, how to calculate the *golden ratio* number 1.618033988749894... using 2 consecutive Fibonacci numbers, as `φ[N] = F[N]/F[N-1]` and `Φ = Φ[Ν]` as `N -> inf`.
+
+```
+#include <math.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#define MAX 50
+
+static int fibonacci[MAX];
+
+void calculate_fibonacci(size_t N) {
+    fibonacci[0] = 0;
+    fibonacci[1] = 1;
+
+    for (int i = 2; i < N; ++i) {
+        fibonacci[i] = fibonacci[i-2] + fibonacci[i-1];
+    }
+}
+
+void print_fibonacci(size_t N) {
+    for (int i = 0; i < N; ++i) {
+        printf("%d ", fibonacci[i]);
+    }
+    printf("\n");
+}
+
+float golden_ratio(int N) {
+    float golden_ratio = fibonacci[N]/fibonacci[N-1];
+    return golden_ratio;
+}
+
+int main() {
+    calculate_fibonacci(100);
+    print_fibonacci(100);
```

Comment:
This can't be right no? `100` is larger than `MAX` so you're going outside the bounds of the array aren't you?